diff --git a/cuda/sample_consensus/src/sac_model_1point_plane.cu b/cuda/sample_consensus/src/sac_model_1point_plane.cu
index 5b88565..ab81404 100644
--- a/cuda/sample_consensus/src/sac_model_1point_plane.cu
+++ b/cuda/sample_consensus/src/sac_model_1point_plane.cu
@@ -44,6 +44,7 @@
 #include <thrust/copy.h>
 #include <thrust/count.h>
 #include <thrust/random.h>
+#include <thrust/gather.h>
 
 #include <vector_types.h>
 #include <stdio.h>
diff --git a/cuda/sample_consensus/src/sac_model_plane.cu b/cuda/sample_consensus/src/sac_model_plane.cu
index faa6710..ca43124 100644
--- a/cuda/sample_consensus/src/sac_model_plane.cu
+++ b/cuda/sample_consensus/src/sac_model_plane.cu
@@ -41,6 +41,7 @@
 #include <vector_types.h>
 #include <thrust/copy.h>
 #include <thrust/count.h>
+#include <thrust/gather.h>
 
 #include <stdio.h>
 #include <limits>
